src/tests/classic/run.rs

```diff
@@ -308,6 +308,67 @@ fn test_treehash_constant_embedded_classic() {
     );
 }
 
+#[test]
+fn test_treehash_constant_embedded_fancy_order() {
+    let result_text = do_basic_run(&vec![
+        "run".to_string(),
+        "-i".to_string(),
+        "resources/tests".to_string(),
+        indoc! {"
+            (mod ()
+              (include sha256tree.clib)
+              (defconst C 18)
+              (defconst H (+ C G (sha256tree (q 2 3 4))))
+              (defconst G (+ B A))
+              (defconst A 9)
+              (defconst B (* A A))
+              H
+              )
+        "}
+        .to_string(),
+    ])
+    .trim()
+    .to_string();
+    let result_hash = do_basic_brun(&vec!["brun".to_string(), result_text, "()".to_string()])
```

Comment:
it's not clear to me why you execute the resulting clvm here. Don't you want to ensure that the clvm code has a specific shape? e.g. you want to make sure all the constants are folded down to a single atom. Running the code here makes the test pass even if the constants are not folded.

I suppose one, indirect, way you could test it would be to assert that the cost is low enough to indicate it was just a constant. But it would seem more straight-forward to just check the resulting clvm code from "run".

Reply:
ya i'll put in a test for shape as well.